docs/guide/rolldown.md

```diff
@@ -91,3 +91,92 @@ For real-time discussions and troubleshooting, make sure to join the [Rolldown D
 The `rolldown-vite` package is a temporary solution to gather feedback and stabilize the Rolldown integration. In the future, this functionality will be merged back into the main Vite repository.
 
 We encourage you to try out `rolldown-vite` and contribute to its development through feedback and issue reports.
+
+## Plugin / Framework authors guide
+
+### The list of big changes
+
+- Rolldown is used for build (Rollup was used before)
+- Rolldown is used for the optimizer (esbuild was used before)
+- CommonJS support is handled by Rolldown (@rollup/plugin-commonjs was used before)
+- Oxc is used for syntax lowering (esbuild was used before)
+- lightningcss is used for CSS minification by default (esbuild was used before)
+- Oxc minifier is used for JS minification by default (esbuild was used before)
+- Rolldown is used for bundling the config (esbuild was used before)
+
+### Detecting rolldown-vite
+
+You can detect by either
+
+- checking the `this.meta.rolldownVersion` existence
+
+```js
+const plugin = {
+  resolveId() {
+    if (this.meta.rolldownVersion) {
+      // logic for rolldown-vite
+    } else {
+      // logic for rollup-vite
+    }
+  },
+}
+```
+
+- checking the `rolldownVersion` export existence
+
+```js
+import * as vite from 'vite'
+
+if (vite.rolldownVersion) {
+  // logic for rolldown-vite
+} else {
+  // logic for rollup-vite
+}
+```
+
+### Ignoring option validation in Rolldown
+
+Rolldown throws an error when unknown options or invalid options are passed. Because some options in Rollup are not supported by Rolldown, you may encounter errors.
+
+> Error: Failed validate input options.
+>
+> - For the "preserveEntrySignatures". Invalid key: Expected never but received "preserveEntrySignatures".
+
+To fix this error, you can either
+
+- conditionally pass the option by checking whether it's running with rolldown-vite
+- set `ROLLDOWN_OPTIONS_VALIDATION=loose` to suppress the error
```

Comment:
question: What is preferred here (quick fix vs. "proper solution")?

Reply:
This should be fixed when we switch to rolldown-vite. But until then I think it's not necessary.
For some options that won't be implemented in rolldown before rolldown-vite switch, I think those should be fixed properly now.

Reply:
Let me check the rolldown team if the options would be supported.

Reply:
They told me that the options won't likely to be implemented as-is or soon. So I think the recommendation would be to "conditionally pass the option by checking whether it's running with rolldown-vite".